source/cpu/cpu.go

```diff
@@ -274,33 +274,78 @@ func getCPUModel() map[string]string {
 	return cpuModelInfo
 }
 
-// getHypervisor detects the hypervisor on s390x by reading /proc/sysinfo.
-// If the file does not exist, it returns an empty string with no error.
+// getHypervisor detects the hypervisor on s390x by reading /proc/sysinfo
+// and on x86_64/arm64 using the cpuid library.
+// Returns normalized hypervisor string or "none" for bare metal.
+// Returns "", nil only when hypervisor information is unavailable on s390x
 func getHypervisor() (string, error) {
-	if _, err := os.Stat("/proc/sysinfo"); os.IsNotExist(err) {
-		return "", nil
-	}
+	// use /proc/sysinfo for s390x
+	if _, err := os.Stat("/proc/sysinfo"); err == nil {
+		data, err := os.ReadFile("/proc/sysinfo")
+		if err != nil {
+			return "", err
+		}
 
-	data, err := os.ReadFile("/proc/sysinfo")
-	if err != nil {
+		hypervisor := ""
+		for _, line := range strings.Split(string(data), "\n") {
+			if strings.Contains(line, "Control Program:") {
+				parts := strings.SplitN(line, ":", 2)
+				if len(parts) == 2 {
+					hypervisor = strings.TrimSpace(parts[1])
+				}
+				break
+			}
+		}
+		if hypervisor == "" {
+			return "", nil
+		}
+
+		// Replace forbidden symbols
+		fullRegex := regexp.MustCompile("[^-A-Za-z0-9_.]+")
+		hypervisor = fullRegex.ReplaceAllString(hypervisor, "_")
+		return hypervisor, nil
+	} else if !os.IsNotExist(err) {
 		return "", err
 	}
 
-	hypervisor := "PR/SM"
-	for _, line := range strings.Split(string(data), "\n") {
-		if strings.Contains(line, "Control Program:") {
-			parts := strings.SplitN(line, ":", 2)
-			if len(parts) == 2 {
-				hypervisor = strings.TrimSpace(parts[1])
+	// use cpuid lib for x86/arm
+	if cpuid.CPU.VM() {
+		hv := strings.ToLower(cpuid.CPU.HypervisorVendorID.String())
+	
+		switch hv {
+		case "msvm":
+			return "hyperv", nil
+		case "kvm":
+			return "kvm", nil
+		case "vmware":
+			return "vmware", nil
+		case "xenhvm":
+			return "xen", nil
+		case "bhyve":
+			return "bhyve", nil
+		case "qemu":
+			return "qemu", nil
+		case "apple":
+			return "apple", nil
+		case "qnx":
+			return "qnx", nil
+		case "acrn":
+			return "acrn", nil
+		case "sre":
+			return "sre", nil
```

Comment:
Couldn't we put all these cases where we don't mangle the value under the default case?


```suggestion
		default:
			return hv, nil
```